src/main/java/com/deepoove/swagger/diff/compare/OperationsDiff.java

```diff
@@ -0,0 +1,139 @@
+        param.setVendorExtsFromGroup(paramExtDiff);
+        if (paramExtDiff.vendorExtensionsAreDiff()) {
+          diffResult.setHasContractChanges(true);
+          diffResult.setHasOnlyCosmeticChanges(false);
+        }
+      }
+
+      Property oldResponseProperty = getResponseProperty(oldOperation);
+      Property newResponseProperty = getResponseProperty(newOperation);
+      PropertyDiffResult propertyDiff = propertyDiffer.diff(oldResponseProperty, newResponseProperty);
+      changedOperation.setAddProps(propertyDiff.getIncreased());
+      changedOperation.setMissingProps(propertyDiff.getMissing());
+      changedOperation.setChangedProps(propertyDiff.getChanged());
+
+      changedOperation.setChangeResponseDescription(operationResponseHasCosmeticChanges(oldOperation.getResponses().values(), newOperation.getResponses().values()));
+      changedOperation.setChangeDescription(!Objects.equals(oldOperation.getDescription(), newOperation.getDescription()));
+      changedOperation.setChangeSummary(!Objects.equals(oldOperation.getSummary(), newOperation.getSummary()));
+      changedOperation.setChangeOperationId(!Objects.equals(oldOperation.getOperationId(), newOperation.getOperationId()));
+
+      ChangedExtensionGroup responseExtDiff = extDiffer.diffResGroup(oldOperation.getResponses(), newOperation.getResponses());
+      changedOperation.putSubGroup("responses", responseExtDiff);
+      if (responseExtDiff.vendorExtensionsAreDiff()) {
+        diffResult.setHasContractChanges(true);
+        diffResult.setHasOnlyCosmeticChanges(false);
+      }
+
+      if (changedOperation.isDiff()) {
+        diffResult.getChangedOperations().put(method, changedOperation);
+      }
+      if (!diffResult.hasContractChanges() && changedOperation.hasOnlyCosmeticChanges()) {
+        diffResult.setHasOnlyCosmeticChanges(true);
+      }
+    }
+
+    return diffResult;
+  }
+
+  private static boolean operationResponseHasCosmeticChanges(Collection<Response> oldResponses, Collection<Response> newResponses) {
+    Iterator<Response> oldOpResponseIterator = oldResponses.iterator();
+    Iterator<Response> newOpResponseIterator = newResponses.iterator();
+    boolean isChangeDescription = false;
+
+    while (oldOpResponseIterator.hasNext() && newOpResponseIterator.hasNext()) {
+      String oldDescription = oldOpResponseIterator.next().getDescription();
+      String newDescription = newOpResponseIterator.next().getDescription();
+      if (((oldDescription == null) ^ (newDescription == null)) || ((oldDescription != null) && !oldDescription.equals(newDescription))) {
+        isChangeDescription = true;
+      }
+    }
+
+    return isChangeDescription;
+  }
+
+  private static Property getResponseProperty(Operation operation) {
+    Map<String, Response> responses = operation.getResponses();
+    // temporary workaround for missing response messages
+    if (responses == null) {
+      return null;
```

Comment:
I think there are a bunch of places we could be using `Optional`s, I did not know it was available. I don't think that should be a part of this change though, this snippet only shows up because of refactoring, but it is old code.